apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/logs/page.tsx

```diff
@@ -1,34 +1,57 @@
-import { TableBlankSlate, TableWithHeader } from '@latitude-data/web-ui'
 import {
-  findCommitCached,
-  getDocumentLogsWithMetadataCached,
-} from '$/app/(private)/_data-access'
+  buildPagination,
+  parsePage,
+} from '@latitude-data/core/lib/buildPagination'
+import { computeDocumentLogsWithMetadata } from '@latitude-data/core/services/documentLogs/computeDocumentLogsWithMetadata'
+import { TableBlankSlate, TableWithHeader } from '@latitude-data/web-ui'
+import { findCommitCached } from '$/app/(private)/_data-access'
+import { getCurrentUser } from '$/services/auth/getCurrentUser'
+import { ROUTES } from '$/services/routes'
 
 import { DocumentLogs } from './_components/DocumentLogs'
 
+const PAGE_SIZE = 25
 export default async function DocumentPage({
   params,
+  searchParams,
 }: {
   params: { projectId: string; commitUuid: string; documentUuid: string }
+  searchParams: { [key: string]: string | string[] | undefined }
 }) {
+  const { workspace } = await getCurrentUser()
   const projectId = Number(params.projectId)
   const commitUuid = params.commitUuid
   const commit = await findCommitCached({ projectId, uuid: commitUuid })
-  const logs = await getDocumentLogsWithMetadataCached({
+  const page = parsePage(searchParams.page)
+  const [rows, count] = await computeDocumentLogsWithMetadata({
+    workspaceId: workspace.id,
     documentUuid: params.documentUuid,
-    commit,
+    draft: commit,
+    pagination: { page, pageSize: PAGE_SIZE },
   })
-
+  const baseUrl = ROUTES.projects
+    .detail({ id: projectId })
+    .commits.detail({ uuid: commitUuid })
+    .documents.detail({ uuid: params.documentUuid }).logs.root
+  const pagination = buildPagination({
```

Comment:
I do a pagination object with the URLs for the pages generated with the right path. This way the `<WebPagination />` is agnostic